pkg/controllers/cronfederatedhpa/cronfederatedhpa_job.go

```diff
@@ -0,0 +1,280 @@
+	ctx := context.Background()
+
+	scaleClient := c.client.SubResource("scale")
+
+	targetGV, err := schema.ParseGroupVersion(cronFHPA.Spec.ScaleTargetRef.APIVersion)
+	targetGVK := schema.GroupVersionKind{
+		Group:   targetGV.Group,
+		Kind:    cronFHPA.Spec.ScaleTargetRef.Kind,
+		Version: targetGV.Version,
+	}
+	targetResource := &unstructured.Unstructured{}
+	targetResource.SetGroupVersionKind(targetGVK)
+	err = c.client.Get(ctx, types.NamespacedName{Namespace: cronFHPA.Namespace, Name: cronFHPA.Spec.ScaleTargetRef.Name}, targetResource)
+	if err != nil {
+		klog.Errorf("Get Resource(%s/%s) failed: %v", cronFHPA.Namespace, cronFHPA.Spec.ScaleTargetRef.Name, err)
+		return err
+	}
+
+	scaleObj := &unstructured.Unstructured{}
+	err = scaleClient.Get(ctx, targetResource, scaleObj)
+	if err != nil {
+		klog.Errorf("Get Scale for resource(%s/%s) failed: %v", cronFHPA.Namespace, cronFHPA.Spec.ScaleTargetRef.Name, err)
+		return err
+	}
+
+	scale := &autoscalingv1.Scale{}
+	err = helper.ConvertToTypedObject(scaleObj, scale)
+	if err != nil {
+		klog.Errorf("Convert Scale failed: %v", err)
+		return err
+	}
+
+	if scale.Spec.Replicas != *c.rule.TargetReplicas {
+		if err := helper.ApplyReplica(scaleObj, int64(*c.rule.TargetReplicas), util.ReplicasField); err != nil {
+			klog.Errorf("Apply Replicas for %s/%s failed: %v", cronFHPA.Namespace, cronFHPA.Spec.ScaleTargetRef.Name, err)
+			return err
+		}
+		err := scaleClient.Update(ctx, targetResource, client.WithSubResourceBody(scaleObj))
+		if err != nil {
+			klog.Errorf("Update Scale failed: %v", err)
+			return err
+		}
+		klog.V(4).Infof("Resource(%s/%s) has be scaled successfully", cronFHPA.Namespace, cronFHPA.Spec.ScaleTargetRef.Name)
+		return nil
+	}
+	return nil
+}
+
+func (c *CronFederatedHPAJob) addFailedExecutionHistory(
+	cronFHPA *autoscalingv1alpha1.CronFederatedHPA, errMsg string) error {
+	exists := false
+
+	nextExecutionTime := cronexpr.MustParse(c.rule.Schedule).Next(time.Now())
+	for index, rule := range cronFHPA.Status.ExecutionHistories {
+		if rule.RuleName == c.rule.Name {
+			failedExecution := autoscalingv1alpha1.FailedExecution{
+				ScheduleTime:  rule.NextExecutionTime,
+				ExecutionTime: &metav1.Time{Time: time.Now()},
+				Message:       errMsg,
+			}
+			historyLimits := helper.CronFederatedHPAFailedHistoryLimits(c.rule)
+			if len(rule.FailedExecutions) > historyLimits-1 {
+				rule.FailedExecutions = rule.FailedExecutions[:historyLimits-1]
+			}
+			cronFHPA.Status.ExecutionHistories[index].FailedExecutions =
+				append([]autoscalingv1alpha1.FailedExecution{failedExecution}, rule.FailedExecutions...)
+			cronFHPA.Status.ExecutionHistories[index].NextExecutionTime = &metav1.Time{Time: nextExecutionTime}
+			exists = true
+			break
+		}
+	}
+
+	// If this history not exist, it means the rule is suspended or deleted, so just ignore it.
+	if exists {
+		updateErr := retry.RetryOnConflict(retry.DefaultRetry, func() (err error) {
```

Comment:
When we use `RetryOnConflict` to update the cronFHPA, when updates failed, it may get the newest the cronFHPA and return the updateErr.